mix.exs

```diff
@@ -27,7 +27,7 @@ defmodule PlugSignature.MixProject do
   # Run "mix help deps" to learn about dependencies.
   defp deps do
     [
-      {:plug, "~> 1.5"},
+      {:plug, "~> 1.14"},
```

Comment:
I require at least Plug 1.14.
If someone is still using 1.12 and setting the host header, the changes will not work.
It wouldn't be easy to keep the library backwards compatible.
For something dealing with security, I don't think we should complicate the code with `if old version do that else something else"

Reply:
But everything does work on older Plug versions, no? In particular the core signature verification, which hasn't  really changed. But also tests work fine on older versions.

Even if, in a custom test in a project that uses this `put_req_header(conn, "host", host)` is still used, then the signature generation still works correctly and the test continues to work until the project is updated to Plug 1.14.

Or am I missing something...?

Reply:
I haven't tested that. Maybe `plug` started setting the `conn.host` during the creation of the conn before they introduced the change that does not allow setting it in headers in 1.14.
In that case, I'll change the requirement back to 1.5

Reply:
I did some more checks. I wanted to ensure the `host` is always correctly set in the `Conn` and yes:
- in the current version the host is parsed from the URI https://github.com/elixir-plug/plug/blob/main/lib/plug/conn/adapter.ex#L23
- in the old 1.5 it was set in cowboy adapter: https://github.com/elixir-plug/plug/blob/v1.5.0/lib/plug/adapters/cowboy2/conn.ex#L7